### PR TITLE
ISPN-3226 Maven report generation rework

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-3226

    * Replace FindBugs with SpotBugs
    * Update plugins
    * Add dependency check to default set
    * Enable for nightly CI builds

### CODE REVIEW

**Comment by tristantarrant** [comment]
The reports add 1h to the build time (from 1h36 to 2h39). This would not apply to PR builds.

**Comment by danberindei** [comment]
The server tests hanged in the last built @tristantarrant

File: pom.xml
Comment (danberindei) [inline pom.xml:2073]: I'd put `dependency-check` in the file name.

File: Jenkinsfile
Comment (danberindei) [inline Jenkinsfile:38]: Could we get away without building the site? The HTML output isn't usable as is anyway, we'd need to run the `site:deploy` or `site:stage` goals to put them in the correct directory structure (the links from the parent use the artifact id as the subdirectory name instead of the source directory name).

**Review by danberindei — CHANGES_REQUESTED** [review]

File: build-configuration/pom.xml
Comment (tristantarrant) [reply build-configuration/pom.xml:10]: Done

File: etc/spotbugs-exclusions.xml
Comment (tristantarrant) [reply etc/spotbugs-exclusions.xml]: Sure

File: commons/all/src/main/java/org/infinispan/commons/util/JVMMemoryInfoInfo.java
Comment (tristantarrant) [reply commons/all/src/main/java/org/infinispan/commons/util/JVMMemoryInfoInfo.java]: Fixed

File: pom.xml
Comment (tristantarrant) [reply pom.xml]: Done

File: Jenkinsfile
Comment (tristantarrant) [reply Jenkinsfile]: I'm now just building the individual reports.

**Comment by tristantarrant** [comment]
@danberindei I've addressed all your comments, let's wait for the CI run

**Comment by tristantarrant** [comment]
The reports look good to me now. Before merging I need to adjust Jenkinsfile

**Comment by danberindei** [comment]
Thanks Tristan, merged!

I think there were still quite a few warnings from generated code in build # 9, and I would have preferred to have properties in pom.xml for all the plugin versions, but there's no point in delaying this any more.

**Pull request merged**
